test/cmd/kubeconfig.sh

```diff
@@ -43,6 +43,29 @@ run_kubectl_config_set_tests() {
   set +o errexit
 }
 
+run_kubectl_config_set_jsonpath_tests() {
+  set -o nounset
+  set -o errexit
+
+  create_and_use_new_namespace
+  kube::log::status "Testing kubectl(v1:config set)"
+
+  # Get the api cert and add a comment to avoid flag parsing problems
+  cert_data=$(echo "#Comment" && cat "${TMPDIR:-/tmp}/apiserver.crt")
+
+  kubectl config set '{.clusters[?(@.name == "test-cluster")].cluster.certificate-authority-data}' "$cert_data" --set-raw-bytes
+  r_written=$(kubectl config view --raw -o jsonpath='{.clusters[?(@.name == "test-cluster")].cluster.certificate-authority-data}')
+
+  encoded=$(echo -n "$cert_data" | base64)
+  kubectl config set '{.clusters[?(@.name == "test-cluster")].cluster.certificate-authority-data}' "$encoded"
+  e_written=$(kubectl config view --raw -o jsonpath='{.clusters[?(@.name == "test-cluster")].cluster.certificate-authority-data}')
+
+  test "$e_written" == "$r_written"
```

Comment:
I do not see how this tests that functionality of using jsonpath. The expected output is also a result from using the new jsonpath feature. What if both responses were wrong but equal?

Reply:
Thank you for pointing this out, this was a complete misunderstanding of what the original test I copied this from was doing, I'll update that so that it actually tests something useful.